lib/galaxy/tools/deps/resolvers/conda.py

```diff
@@ -146,15 +146,11 @@ def resolve_all(self, requirements, **kwds):
             if requirement.type != "package":
                 return False
 
-        conda_targets = []
-        for requirement in requirements:
-            requirement = self._expand_requirement(requirement)
-
-            version = requirement.version
-            if self.versionless:
-                version = None
-
-            conda_targets.append(CondaTarget(requirement.name, version=version))
+        expanded_requirements = set([self._expand_requirement(r) for r in requirements])
```

Comment:
Awesome thanks! I'm about to open a PR against your branch that shuffles some other stuff in the PR around quite a bit but it won't touch this line I don't believe.